inference-engine/src/mkldnn_plugin/utils/node_dumper.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#ifdef CPU_DEBUG_CAPS
+#pragma once
+
+#include "mkldnn_node.h"
+
+#include <array>
+#include <string>
+
+namespace MKLDNNPlugin {
+
+/**
+ * Blobs are not dumped by default
+ * Blobs are dumped if node matches all specified env filters
+ *
+ * To dump blobs from all the nodes use the following filter:
+ *
+ * OV_CPU_BLOB_DUMP_NODE_NAME=.+
+ */
+class NodeDumper {
+public:
+    NodeDumper(int _count) : count(_count) {
+        setup();
+    }
+
+    void setup();
+
+    void dumpInputBlobs(const MKLDNNNodePtr &node) const;
+
+    void dumpOutputBlobs(const MKLDNNNodePtr &node) const;
+
+private:
+    void dumpInternalBlobs(const MKLDNNNodePtr& node) const;
+    bool shouldBeDumped(const MKLDNNNodePtr &node) const;
+
+    bool shouldDumpAsText = false;
+    bool shouldDumpInternalBlobs = false;
+    int count;
+
+    std::string dumpDirName = "mkldnn_dump";
+
+    enum FILTER {
```

Comment:
Considering how it is used, it would be better to use `enum class`.

Reply:
As discussed, we cannot use enum class here since it is used as vector indexes.

Reply:
Agree